### PR TITLE
Product database

### PR DESCRIPTION
**Pull request by MansourWolou**
Creation of table and connexion to database

Linked #39

### CODE REVIEW

File: Server/shop/src/main/java/com/cashmanager/server/shop/Product.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+@Entity
+@Table(name = "products")
+public class Product {
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    @Column( nullable = false)
+    private UUID product_id;
+    private String name;
+    private Integer price;
+    private String productUrl;
```
Comment (MelvinCou) [inline Server/shop/src/main/java/com/cashmanager/server/shop/Product.java:27]: Add `@Column( nullable = false)` for each field
non of theme are nullable

File: Server/shop/src/main/java/com/cashmanager/server/shop/Order.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+@Builder
+@AllArgsConstructor
+@Entity
+@Table(name = "orders")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    @Column( nullable = false)
+    // GenerationType instructs that a UUID for the entity should be generated automatically for us by the persistence provider.
+    private UUID order_id;
```
Comment (MelvinCou) [inline Server/shop/src/main/java/com/cashmanager/server/shop/Order.java:25]: We already know that we are in an Order, this is not a foreign key

```suggestion
    private UUID id;
```

File: Server/shop/src/main/java/com/cashmanager/server/shop/Order.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    private String status;
+    private Date date;
+
+    @OneToMany(mappedBy = "orders")
+    @ToString.Exclude
+    private Set<Ordered_orders> ordered_orders;
```
Comment (MelvinCou) [inline Server/shop/src/main/java/com/cashmanager/server/shop/Order.java:31]:
```suggestion
    private Set<Ordered_orders> orderedOrders;
```

File: Server/shop/src/main/java/com/cashmanager/server/shop/Ordered_orders.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.Objects;
+import java.util.UUID;
+@Getter
+@Setter
+@ToString
+@RequiredArgsConstructor
+@Builder
+@AllArgsConstructor
+@Entity
+public class Ordered_orders {
```
Comment (MelvinCou) [inline Server/shop/src/main/java/com/cashmanager/server/shop/Ordered_orders.java:16]:
```suggestion
public class OrderedOrders {
```

File: Server/shop/src/main/java/com/cashmanager/server/shop/Ordered_orders.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    @Column( nullable = false)
+    private UUID ordered_orders_id;
```
Comment (MelvinCou) [inline Server/shop/src/main/java/com/cashmanager/server/shop/Ordered_orders.java:20]:
```suggestion
    private UUID id;
```

**Review by MelvinCou — CHANGES_REQUESTED** [review]

File: Server/shop/src/main/java/com/cashmanager/server/shop/Order.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+@Builder
+@AllArgsConstructor
+@Entity
+@Table(name = "orders")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    @Column(name = "order_id" ,nullable = false)
+    // GenerationType instructs that a UUID for the entity should be generated automatically for us by the persistence provider.
+    private UUID orderId;
```
Comment (MelvinCou) [inline Server/shop/src/main/java/com/cashmanager/server/shop/Order.java:25]: Please stay simple, use only `table_id` naming for foreign keys.
Also do that for the other tables.

```suggestion
    private UUID id;
```

**Review by MelvinCou — CHANGES_REQUESTED** [review]

File: Server/shop/src/main/java/com/cashmanager/server/shop/Order.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+@ToString
+@RequiredArgsConstructor
+@Builder
+@AllArgsConstructor
+@Entity
+@Table(name = "orders")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    @Column(name = "order_id" ,nullable = false)
```
Comment (MeloLawson) [inline Server/shop/src/main/java/com/cashmanager/server/shop/Order.java:23]: for the primary key, maybe name it only "id" and not "order_id"

File: Server/shop/src/main/java/com/cashmanager/server/shop/Order.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    private UUID orderId;
+    @Column( nullable = false)
+    private String status;
+    @Column( nullable = false)
+    private Date date;
+
+    @OneToMany(mappedBy = "orders")
+    @ToString.Exclude
+    private Set<OrderedOrder> ordered_orders;
+
```
Comment (MeloLawson) [inline Server/shop/src/main/java/com/cashmanager/server/shop/Order.java:34]: be careful on the field naming, "orderedOrders" and not "ordered_orders"
```suggestion
private Set<OrderedOrder> orderedOrders;
```

File: Server/shop/src/main/java/com/cashmanager/server/shop/OrderedOrder.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+@RequiredArgsConstructor
+@Builder
+@AllArgsConstructor
+@Entity
+@Table(name = "ordered_orders")
+public class OrderedOrder {
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    @Column( nullable = false)
+    private UUID orderedOrdersId;
```
Comment (MeloLawson) [inline Server/shop/src/main/java/com/cashmanager/server/shop/OrderedOrder.java:22]: same remark for the primary key name
```suggestion
    private UUID id;
```

File: Server/shop/src/main/java/com/cashmanager/server/shop/OrderedOrder.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    @Column( nullable = false)
+    private Integer quantity;
+
+    @ManyToOne
+    @JoinColumn(name = "product_id" ,nullable=false)
+    private Product products;
```
Comment (MeloLawson) [inline Server/shop/src/main/java/com/cashmanager/server/shop/OrderedOrder.java:28]: There is only one product, so the field should not be plural
```suggestion
    private Product product;
```

File: Server/shop/src/main/java/com/cashmanager/server/shop/OrderedOrder.java [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    @ManyToOne
+    @JoinColumn(name = "order_id", nullable=false)
+    private Order orders;
```
Comment (MeloLawson) [inline Server/shop/src/main/java/com/cashmanager/server/shop/OrderedOrder.java:31]: There is only one order, so the field should not be plural
```suggestion
    private Order order;
```

File: Server/shop/src/main/java/com/cashmanager/server/shop/Product.java [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+@RequiredArgsConstructor
+@Builder
+@AllArgsConstructor
+@Entity
+@Table(name = "products")
+public class Product {
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    @Column( name = "product_id",nullable = false)
+    private UUID productId;
```
Comment (MeloLawson) [inline Server/shop/src/main/java/com/cashmanager/server/shop/Product.java:24]:
```suggestion
    private UUID id;
```

File: Server/shop/src/main/java/com/cashmanager/server/shop/Product.java [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+    @Column( nullable = false)
+    private Integer price;
+    @Column( nullable = false)
+    private String productUrl;
+    @Column( nullable = false)
+    private Integer stock;
+
+    @OneToMany(mappedBy="products")
+    @ToString.Exclude
+    private Set<OrderedOrder> ordered_orders;
```
Comment (MeloLawson) [inline Server/shop/src/main/java/com/cashmanager/server/shop/Product.java:36]:
```suggestion
    private Set<OrderedOrder> orderedOrders;
```

**Review by MeloLawson — CHANGES_REQUESTED** [review]

File: Server/shop/src/main/java/com/cashmanager/server/shop/OrderedOrder.java [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+@RequiredArgsConstructor
+@Builder
+@AllArgsConstructor
+@Entity
+@Table(name = "ordered_orders")
+public class OrderedOrder {
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    @Column( name = "id",nullable = false)
+    private UUID orderedOrderId;
```
Comment (MelvinCou) [inline Server/shop/src/main/java/com/cashmanager/server/shop/OrderedOrder.java:22]:
```suggestion
    private UUID Id;
```

**Review by MelvinCou — APPROVED** [review]

**Review by MeloLawson — APPROVED** [review]

**Review by MelvinCou — APPROVED** [review]

**Pull request merged**